CMakeLists.txt

```diff
@@ -0,0 +1,198 @@
+
+if (NOT WIN32)
+  cmake_minimum_required(VERSION 3.7.0)
+else ()
+  cmake_minimum_required(VERSION 3.8.0)
```

Comment:
Can we just stick with 3.8?

Reply:
we want to be consistent with mesos build. it uses 3.7.0 for linux.